### DISCUSSION TITLE
It's possible have two forms with the same configuration?

### DISCUSSION THREAD

**Discussion by carl0smat3us** [react-hook-form/react-hook-form#11549] [Q&A] [answered]
The Submit funtion is called only in the second form.

I just tried create two onSumbit function.

```tsx
'use client'

import Link from 'next/link'
import { MoveLeft, Search, ShoppingCart } from 'lucide-react'
import { useState } from 'react'
import clsx from 'clsx'
import * as z from 'zod'
import { zodResolver } from '@hookform/resolvers/zod'
import useCartStore from '@/store/CartStore'
import { useForm } from 'react-hook-form'
import { useRouter } from 'next/navigation'
import Image from 'next/image'

interface ClientHeaderProps {
  searchDefault?: string | null
}

const schema = z.object({
  searchValue: z.string().min(1),
})

interface FormData {
  searchValue: string
}

export default function ClientHeader(props: ClientHeaderProps) {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FormData>({
    resolver: zodResolver(schema),
  })
  const cartProducts = useCartStore((state) => state.products)
  const [isSearchOn, setSearchOn] = useState(false)
  const router = useRouter()

  function onSubmit(data: FormData) {
    router.push(`/search/${data.searchValue}`)
  }

  return (
    <header className="border-b">
      <article
        className={clsx(
          'w-screen flex sm:flex justify-between items-center px-4 py-4',
          {
            hidden: isSearchOn,
          },
        )}
      >
        <Link href="/">
          <Image
            src="/logo.png"
            alt="Any logo"
            width={90}
            height={90}
          />
        </Link>

        <div className="flex gap-4">
          <form
            onSubmit={handleSubmit(onSubmit)}
            className="max-sm:hidden h-11 w-72 flex justify-between items-center bg-neutral-100"
          >
            <input
              type="text"
              placeholder="?"
              defaultValue={props.searchDefault || ''}
              {...register('searchValue')}
              className={`pl-4 h-full w-[85%] bg-transparent outline-none border-l border-t border-b border-transparent text-black placeholder:text-sm placeholder:text-[#303030] focus:border-main ${errors.searchValue && 'border-red-500'}`}
            ></input>
            <button
              className="bg-main h-full w-[15%] flex items-center justify-center border-main"
              type="submit"
            >
              <Search color="#fff" size={18} />
            </button>
          </form>
          <div className="h-11 flex gap-4 items-center justify-between">
            <button className="hidden max-sm:block">
              <Search
                onClick={() => setSearchOn(true)}
                color="#000"
                size={27}
              />
            </button>
            <Link
              href="/cart"
              className="inline-flex relative justify-center w-full border border-gray-300 shadow-sm p-2 rounded-full bg-white text-sm font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 focus:ring-offset-gray-100"
            >
              <ShoppingCart color="#000" size={27} />
              <div className="absolute bottom-7 left-7 bg-black rounded-full flex justify-center items-center w-6 h-6">
                <span
                  className={clsx(
                    'text-white text-[12px] text-center font-semibold',
                    {
                      hidden: cartProducts.length >= 10,
                    },
                  )}
                >
                  {cartProducts.length}
                </span>
              </div>
            </Link>
          </div>
        </div>
      </article>

      <article
        className={clsx('items-center justify-around flex sm:hidden', {
          hidden: !isSearchOn,
        })}
      >
        <button
          className="h-full w-[10%] flex items-center justify-center"
          onClick={() => setSearchOn(false)}
        >
          <MoveLeft color="#000" size={20} />
        </button>

        <form
          onSubmit={handleSubmit(onSubmit)}
          className="flex justify-between items-center m-auto h-11 my-4 w-[80%] bg-neutral-100 border-b"
        >
          <input
            type="text"
            placeholder="?"
            defaultValue={props.searchDefault || ''}
            {...register('searchValue')}
            className={`pl-4 h-full w-[85%] bg-transparent outline-none border-l border-t border-b border-transparent text-black placeholder:text-sm placeholder:text-[#303030] focus:border-main ${errors.searchValue && 'border-red-500'}`}
          ></input>
          <button
            type="submit"
            className="bg-main h-full w-[15%] flex items-center justify-center border-main"
          >
            <Search color="#fff" size={18} />
          </button>
        </form>
      </article>
    </header>
  )
}
```

**Comment by Moshyfawn** [comment, marked as answer]
I am uncertain about your use case. If you register multiple inputs with the same name _(`searchValue`)_, only the last input will be registered, overriding any previous inputs with the same name.

In [my sandbox](https://codesandbox.io/p/sandbox/tender-panka-kpxmjw?file=%2Fsrc%2FApp.tsx%3A13%2C23&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt27g9ec00063b6enhu7eom1%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt27g9ec00023b6ek51zhiz1%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt27g9ec00033b6eeocmokcs%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt27g9ec00053b6el796ovdg%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt27g9ec00023b6ek51zhiz1%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt27g9ec00013b6e822kgxho%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clt27kx6k00023b6ebcxs57b2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A13%252C%2522startColumn%2522%253A23%252C%2522endLineNumber%2522%253A13%252C%2522endColumn%2522%253A23%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clt27g9ec00023b6ek51zhiz1%2522%252C%2522activeTabId%2522%253A%2522clt27kx6k00023b6ebcxs57b2%2522%257D%252C%2522clt27g9ec00053b6el796ovdg%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt27g9ec00043b6elrwsfhvc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clt27g9ec00053b6el796ovdg%2522%252C%2522activeTabId%2522%253A%2522clt27g9ec00043b6elrwsfhvc%2522%257D%252C%2522clt27g9ec00033b6eeocmokcs%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt27g9ec00033b6eeocmokcs%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D), I can submit both forms with the same submission handler, unless you mean some other behaviour.
